packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -928,14 +933,22 @@ Future<T> showCupertinoModalPopup<T>({
   @required BuildContext context,
   @required WidgetBuilder builder,
   ImageFilter filter,
+  Color barrierColor,
   bool useRootNavigator = true,
   bool semanticsDismissible,
 }) {
   assert(useRootNavigator != null);
+
+  barrierColor = barrierColor ?? CupertinoDynamicColor.resolve(_kModalBarrierColor, context);
+
+  if (barrierColor.alpha == 0) {
```

Comment:
ah I see your dilemma. This is indeed gonna make it pretty obscure :)

One suggestion: just make `barrierColor` default to `_kModalBarrierColor`. Then later resolve barrierColor with context. So if the user doesn't touch it, it has the default behavior. If the user sets it to null, it resolves to null also. If the user sets a custom Color or CupertinoDynamicColor, then they get that custom color.